src/main/java/com/staxrt/tutorial/controller/PostController.java

```diff
@@ -0,0 +1,73 @@
+package com.staxrt.tutorial.controller;
+/* https://www.baeldung.com/spring-data-jpa-multiple-databases */
+
+import com.staxrt.tutorial.exception.ResourceNotFoundException;
+import com.staxrt.tutorial.model.Post;
+import com.staxrt.tutorial.model.User;
+import com.staxrt.tutorial.repository.PostRepository;
+import com.staxrt.tutorial.repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@RestController
+@RequestMapping("/api/v2")
+
+public class PostController {
+
+    @Autowired
+    private PostRepository postRepository;
+
+    @GetMapping("/posts")
+    public List<Post> getAllPosts() {
+        return postRepository.findAll();
+    }
+
+    @GetMapping("/posts/{id}")
+    public ResponseEntity<Post> getPostById(@PathVariable(value = "id") Long postId)
+            throws ResourceNotFoundException {
+        Post post =
+                postRepository
+                        .findById(postId)
+                        .orElseThrow(() -> new ResourceNotFoundException("Post not found on :: " + postId));
+        return ResponseEntity.ok().body(post);
+    }
+    @PostMapping("/posts")
+    public Post createPost(@Valid @RequestBody Post post) {
+        return postRepository.save(post);
+    }
+    @PutMapping("/posts/{id}")
+    public ResponseEntity<Post> updatePost(
+            @PathVariable(value = "id") Long postId, @Valid @RequestBody Post postDetails)
```

Comment:
Aktualnie masz tak, że po stronie klienta ktoś będzie mógł nadpisać datę modyfikacji co nie jest zbyt fajne. Data modyfikacji powinna być ustalana po stronie serwera. Dlatego nie powinno się encji przepychać w kontrollerach. Powinieneś mieć oddzielną warstwę data transfer object, w której decydujesz co będzie dostępne przez api. Potem te DTO przepisujesz na encję (twój model bazodanowy) i dopiero po przepisaniu zapisujesz do bazy. Niestety to wiąże się ze sporą duplikacją tych klas, ale nie ma za bardzo cwanego sposobu jak to inaczej rozwiązać. Są jakieś koncepty jak modelmapper opisany tutaj: https://www.baeldung.com/entity-to-and-from-dto-for-a-java-spring-application , ale za bardzo tego nie używałem.

Reply:
TODO dzisiaj